### PR TITLE
[BugFix][Frontend] Use correct, shared tokenizer in OpenAI server

### PR DESCRIPTION
**Pull request by njhill**
The front-end server code currently doesn't use lora-specific tokenizers.

It also won't make use of the recently introduced parallel async tokenization if enabled.

### CODE REVIEW

**Comment by njhill** [comment]
Test failures look unrelated (network blips).

**Review by Yard1 — COMMENTED** [review]
Could we add a test? We can mock some stuff - just to make sure that if we go through the OpenAI server with different lora requests, they are tokenized correctly.

**Comment by DarkLight1337** [comment]
Can the same tokenizer be used to apply the chat template as well?

File: vllm/engine/async_llm_engine.py [hunk 1]
```diff
@@ -378,6 +379,12 @@ def _error_callback(self, exc: Exception) -> None:
         self.set_errored(exc)
         self._request_tracker.propagate_exception(exc)
 
+    async def get_tokenizer_group(self) -> BaseTokenizerGroup:
+        if self.engine_use_ray:
+            return await self.engine.get_tokenizer_group.remote()
+        else:
+            return self.engine.get_tokenizer_group()
```
Comment (dtrifiro) [inline vllm/engine/async_llm_engine.py:386]: nit: 

```suggestion
        return self.engine.get_tokenizer_group()
```

**Comment by DarkLight1337** [comment]
Closing as superseded by #6227.

**Pull request closed without merging**